### PR TITLE
[8.1] [DOCS] Remove JWT secure settings

### PR DESCRIPTION
**Pull request by lockewritesdocs**
The JWT secure settings were added to `8.1` in #81680, but the code changes in 8.1 were internal and cannot be used in that release. This PR removes those settings.

### CODE REVIEW

File: docs/reference/settings/security-settings.asciidoc [hunk 1]
```diff
@@ -1636,19 +1636,10 @@ at the OpenID Connect Provider.
 `rp.client_auth_method` {ess-icon}::
 (<<static-cluster-setting, Static>>)
 The client authentication method used by {es} to authenticate
-to the OpenID Connect Provider. Can be `client_secret_basic`, `client_secret_post`,
-or `client_secret_jwt`. Defaults to `client_secret_basic`.
+to the OpenID Connect Provider. Can be `client_secret_basic` or 
+`client_secret_post`. Defaults to `client_secret_basic`.
```
Comment (justincr-elastic) [inline docs/reference/settings/security-settings.asciidoc:1640]: JWT references should not be removed from OIDC realm. OIDC realm supports JWTs too.

File: docs/reference/settings/security-settings.asciidoc [hunk 2]
```diff
@@ -1636,19 +1636,10 @@ at the OpenID Connect Provider.
...
 // end::rp-client-auth-method-tag[]
 
-// tag::rp-client-auth-jwt-signature-algorithm[]
```
Comment (justincr-elastic) [inline docs/reference/settings/security-settings.asciidoc:1643]: JWT references should not be removed from OIDC realm. OIDC realm supports JWTs too.

**Review by justincr-elastic — CHANGES_REQUESTED** [review]

File: docs/reference/settings/security-settings.asciidoc (hunk 1 above)
Comment (lockewritesdocs) [reply docs/reference/settings/security-settings.asciidoc]: Thanks for the clarification -- I reverted this change.

File: docs/reference/settings/security-settings.asciidoc (hunk 2 above)
Comment (lockewritesdocs) [reply docs/reference/settings/security-settings.asciidoc:1643]: Added this setting back as well.

**Comment by lockewritesdocs** [comment]
@justincr-elastic, let me know if this PR is good to merge now. If so, I can merge and then backport your changes for #86746.

**Pull request merged**